src/include/miopen/solver.hpp

```diff
@@ -1204,6 +1204,77 @@ struct ConvAsmImplicitGemmV4R1DynamicBwd : SolverBase<ConvolutionContext>
     ConvSolution GetSolution(const ConvolutionContext&) const;
 };
 
+struct TunableImplicitGemmGTCDynamic_t
+{
+    std::string direction;
+    std::string precision;
+    int nxb;
+    int nxe;
+
+    int gemm_m_per_block;
+    int gemm_n_per_block;
+    int gemm_k_per_block;
+
+    int wave_tile_m;
+    int wave_tile_n;
+    int wave_step_m;
+    int wave_step_n;
+    int wave_repeat_m;
+    int wave_repeat_n;
+
+    int tensor_a_thread_lengths[4];
+    int tensor_a_cluster_lengths[4];
+    int tensor_b_thread_lengths[4];
+    int tensor_b_cluster_lengths[4];
+    int gemm_k_global_split;
+
+    int GetBlockSize()
+    {
+        const auto WaveSize = 64;
+        auto block_size     = (gemm_m_per_block / (wave_tile_m * wave_step_m * wave_repeat_m)) *
+                          (gemm_n_per_block / (wave_tile_n * wave_step_n * wave_repeat_n)) *
+                          WaveSize;
+        return block_size;
+    }
+
+    std::string GetKernelName()
+    {
+        std::string kernel_name = std::string("igemm_");
```

Comment:
Use `std::ostringstream` here, please.

Reply:
@DrizztDoUrden this is change, and use the same code as in https://github.com/ROCmSoftwarePlatform/MIOpen/pull/442 (merged PR).